### ISSUE TITLE
Docs: Example thumbnails in docs

### ISSUE THREAD

**Issue by kolibril13** [ManimCommunity/manim#269]
Ideas for the documentation page:

The way I learned manim, was with lots of examples.
As this might be very useful for others as well, it would be lovely when the documentation of the classes and methods have easily accessible links to examples.
I looked into the documentation of matplotlib, and there they have under each explanation thumbnails of examples, where this or that class/method is used.
See for example here:
https://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.boxplot.html#matplotlib.axes.Axes.boxplot
When you click on one of the example thumbnails, you will be directed to the corresponding example with code and the output image.

In the case of manim, we can have this as well. 
What do you think about this?

**Comment by leotrs** [comment]
Yes, I like what matplotlib does with examples in general. One thing I don't like is that they have _so many_ examples that some of them are quite a bit better (more illustrative, clearer, better explained) than others. I presume that's just a byproduct of being such a huge library.

In any case, I'm in favor of this. HOWEVER, I suggest we start with something basic, and iterate from there. The system that matplotlib uses makes sense for them because they have dozens (hundreds?) of examples. I don't think it's worth to build this whole system for us at this stage.

Let's just focus on having a single examples page in the documentation with links to a few examples. When we have that, we can talk about how to connect it to the rest of the documentation and so on.

**Comment by leotrs** [comment]
@kolibril13 you should look into this https://sphinx-gallery.github.io/stable/index.html for building the examples
